utils/firestore.js

```diff
@@ -1410,136 +1410,63 @@ const getSpecimenCollections = async (token, siteCode) => {
+    const trackingId = filters.trackingId ?? '';
+    const endDate = filters.endDate ?? '0';
 
-    let startDate = "0";
-    let trackingId = '';
-    let endDate = "0";
-    if(filters.hasOwnProperty('startDate')){
-        startDate = filters['startDate']
-    }
-    
-    if(filters.hasOwnProperty('trackingId')){
-        trackingId = filters['trackingId'];
-    }
-    if(filters.hasOwnProperty('endDate')){
-        endDate = filters['endDate']
-    }
-    let snapshot = {'docs':[]};
-    if(trackingId !== ''){ 
-        if(endDate !== "0"){ 
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).where('656548982', '<=', endDate).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).where('656548982', '<=', endDate).orderBy('656548982', 'desc').get();
-            }
-      }
-      else{
-          if(startDate !== "0"){
-              snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-          }
-          else{
-              snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).orderBy('656548982', 'desc').get();
-          }
-      }
-    }
-    else{
-        if(endDate !== "0"){
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', '<=', endDate).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', '<=', endDate).orderBy('656548982', 'desc').get();
-            }
-        }
-        else{
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).orderBy('656548982', 'desc').get();
-            }
-        }
-    }
+    let query = db.collection('boxes').where('145971562', '==', 353358909);
+
+    query = buildQueryWithFilters(query, trackingId, endDate, startDate, source, siteCode)
+
+    const snapshot = await query.orderBy('656548982', 'desc').get();
```

Comment:
Question: I'm not sure we need `orderBy()` since we're just returning `.length` but maybe I'm missing something.

Reply:
orderBy() is a great, efficient implementation when sorting is needed.

Might be worth experimenting with .count() in the future. I haven't used this, but it could be very fast compared to pulling data and counting length: Doc: https://cloud.google.com/firestore/docs/query-data/aggregation-queries